src/v/model/fundamental.h

```diff
@@ -107,6 +107,12 @@ inline bool is_materialized_topic(const model::topic& t) {
     return make_materialized_topic(t).has_value();
 }
 
+inline model::topic materialized_or_source_topic(const model::topic& topic) {
```

Comment:
This name is a little bit confusing to me. It always get the source topic. Maybe just `get_source_topic` would be fine ?

Reply:
+1